src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1877,10 +1875,7 @@ WorkerAggregateExpressionList(Aggref *originalAggregate,
 		CountDistinctErrorRate == DISABLE_DISTINCT_APPROXIMATION &&
 		walkerContext->repartitionSubquery)
 	{
-		Aggref *aggregate = (Aggref *) copyObject(originalAggregate);
-		List *aggTargetEntryList = aggregate->args;
-		TargetEntry *distinctTargetEntry = (TargetEntry *) linitial(aggTargetEntryList);
-		List *columnList = pull_var_clause_default((Node *) distinctTargetEntry);
+		List *columnList = pull_var_clause_default((Node *) originalAggregate);
```

Comment:
This approach is safe since we only return copied columns. However, After going through your comment I have found a simplification on the code.